### PR TITLE
Setup publiopti

### LINKED ISSUE #1716: Documenter et refactorer `rulesToJSON` script
Ça vaudrait le coup de décrire en quelques phrases (dans le wiki ou bien en commentaire) l'algo dans les grandes lignes. En particulier la gestion de la trad/internationalisation avec tous les différents fichiers et à quoi ils correspondent.

De plus, `rulesToJSON` commence à être assez gros, il serait bien de faire une petite passe de refactoring pour en dégager un seul niveau de logique et avoir une suite d'appel de fonction du style :

```js
const baseRules = getBaseRules(...)
const localisedRules = getLocalisedRules(baseRules, regions)
const translatedLocalisedRules =
  getTranslatedLocalisedRules(localisedRules, destLangs)
const optimizedTranslatedLocalisedRules =
  getOptimizedTranslatedLocalisedRules(translatedLocalisedRules)

writeLocalisedRules(localisedRules)
writeTranslatedRules(translatedLocalisedRules)
writeOptimizedRules(optimizedTranslatedLocalisedRules)
```

### PR DESCRIPTION
**Pull request by EmileRolley**
Setup the model optimisation with `publiopti` achieving a ~65% reduction in the number of rules.

## Remarks

When working on the rules optimisation, I discovered empty rules inside the base model such as `pétrole` or `logement . gaz`. Deleting them doesn't seem to be a problem for the simulation.
As full rules are loaded to render documentation page, it isn't an issue to remove them from the optimized model, right? @Clemog

## Changelog

* No standalone `somme` mechanisms in the `empreinte nationale.yaml` file.
* `icone` -> `icônes` in `alimentation.yaml`
* New predicate keeping specified empty rules and ones which contains the `icônes` attribute from the optimisation.
* Upgrade `publicodes` to the `1.0.0-beta.66` version.
* First pass of refactoring for the `rulesToJSON` script -- closes #1716. 
* Support optimisation for localized models.
* Add `yarn test:optim` to compare personas results between the optim and the base model.

### CODE REVIEW

File: scripts/rulesToJSON.mjs [hunk 1]
```diff
@@ -174,63 +168,48 @@ glob(srcFile, { ignore: ['data/i18n/**'] }, (_, files) => {
...
-				outputJSONPath,
-				`co2-model.${region.toUpperCase()}-lang.fr.json`
-			)
-			const regionRuleAttrs =
-				utils.readYAML(path.resolve(`data/i18n/models/${region}.yaml`)) ?? {}
-			const rehydratedRules = addRegionToBaseRules(baseRules, regionRuleAttrs)
-			writeRules(rehydratedRules, destPath, region)
-		})
-
+		destLangs.push(srcLang)
```
Comment (Clemog) [inline scripts/rulesToJSON.mjs:177]: Pas possible de faire en sorte que destLangs contienne 'fr' par défaut ailleurs ? Dans utils ?

Comment (EmileRolley) [reply scripts/rulesToJSON.mjs:177]: Hmm, ouais pourquoi pas. Après vu que le `cli.getArgs` est utilisée par tous les autres scripts il faudrait s'assurer que ça ne pose pas de problème..

File: scripts/modelOptim.mjs [hunk 2]
```diff
@@ -1,15 +1,67 @@
...
+// 				[publiopti] to be used in the build scripts: rulesToJSON.
+//
+// [publiopti]: https:github.com/EmileRolley/publiopti
+
 import Engine from 'publicodes'
 import path from 'path'
 import { readFileSync, writeFileSync } from 'fs'
 import { constantFolding, disabledLogger, getRawNodes } from 'publiopti'
 
+// Rule names which should be kept in the optimized model.
```
Comment (laem) [inline scripts/modelOptim.mjs:11]: Can you shortly explain in the comment why we need this ?

**Comment by laem** [comment]
@EmileRolley what kind of review do you need from me, if any ?

**Comment by EmileRolley** [comment]
> @EmileRolley what kind of review do you need from me, if any ?

A basic code review I think, and some checks on the preview to make sure I haven't forgotten any rules.

**Pull request merged**